src/aideck/core/tools.py

```
@@ -0,0 +1,208 @@
+    def _validate_signature(self):
+        """Validate that the function signature matches the parameters."""
+        sig = signature(self.func)
+        param_names = set(sig.parameters.keys())
+        schema_params = set(self.parameters.get("properties", {}).keys())
+        
+        # Check for missing parameters
+        missing = schema_params - param_names
+        if missing:
+            raise ToolError(
+                f"Function '{self.name}' is missing parameters: {', '.join(missing)}"
+            )
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)